src/flowcean/transforms/resample.py

```diff
@@ -0,0 +1,104 @@
+import logging
+import math
+from typing import cast, override
+
+import numpy as np
+import polars as pl
+from scipy.interpolate import CubicSpline
+
+from flowcean.core import Transform
+from flowcean.utils import is_timeseries_feature
+
+logger = logging.getLogger(__name__)
+
+
+class Resample(Transform):
+    """Resample time series features to a given sampling rate."""
+
+    def __init__(
+        self,
+        sampling_rate: float | dict[str, float],
+        *,
+        use_cubic_interpolation: bool = False,
```

Comment:
Maybe do not use `bool` but rather a string selecting the different methods.

Reply:
Sure, I changed it to a string type. Since we don't use enums anywhere else, I didn't feel like introducing them here.